src/visualizers/Chaos.ts

```diff
@@ -111,15 +116,15 @@ class Chaos extends P5Visualizer {
         },
         highlightWalker: {
             value: this.highlightWalker,
-            forceType: 'integer',
+            type: ParamType.INTEGER,
             displayName: 'Number of walker to highlight',
             required: false,
             visibleDependency: 'colorStyle',
             visibleValue: ColorStyle.Highlighting_one_walker,
         },
         first: {
-            value: '' as string | number,
-            forceType: 'integer',
+            value: 0,
+            type: ParamType.INTEGER,
```

Comment:
It appears you have changed the behavior of the "first" parameter: it is intended to start empty, to mean "use the first available index of the sequence". It seems that with the new code, it will start with 0, which is not the intention: sequences may start with negative or positive indices, and we don't want to override that to arbitrarily start from 0 unless that is the specific intention of the visitor.

Reply:
(Please restore the previous behavior.)

Reply:
Since the forced type of the parameter is `integer`, the parameter system actually won't allow an empty input to pass validation, as this is not technically an integer. So if the parameter type is supposed to accommodate for empty values, maybe this is a new type we'll have to add support for (like `OPTIONAL_INTEGER`). Alternatively, this could be broken into two parameters like `useFirstAvailableIndex`, where with a value of false, the `first` parameter shows up underneath.

Reply:
OK, in terms of the user experience, we definitely need a default blank box here meaning "just use the first index from the sequence", rather than a two-parameter system; the latter would be more cumbersome in practice. I.e., we need the pre-PR behavior. I can think of at least four ways to go here with the new parameter system:

A) Implement an OPTIONAL_INTEGER parameter type as you suggest; the main drawback with that plan is it seems like noticeable complication for something pretty specific and could lead to possible future code duplication, e.g. if someone wants OPTIONAL_COLOR

B) Every parameter for which the "required" property is false really has underlying type `T | undefined` and blank input means `undefined`. Adopting this strategy will definitely make the TypeScript typing more complicated, however, although likely still feasible with the correct type-level language gymnastics.

C) Every parameter for which the "required" property is false must also have a "default" property, and for such parameters, an empty input is explicitly allowed and means the value of the "default" property. 

D) NUMBER and INTEGER parameters explicitly allow empty input meaning NaN, and it is up to the visualizer writer to check against this case in their parameter-specific "validate()" if they don't want NaNs.

Strategy (C) seems in my opinion the cleanest/most straightforward from the visualizer-writer's point of view (indeed, what else does required == `false` mean?), but note that all possibilities on the above list are acceptable, as could well be some other possibility you might propose to allow the original behavior of parameters such as this one. So please choose a plan that seems best from your point of view and include it in this PR. Thanks!